kakarot_scripts/deployment/dualvm_token_deployments.py

```diff
@@ -0,0 +1,188 @@
+# %% Imports
+import json
+import logging
+from typing import Any, Dict, List
+
+from uvloop import run
+
+from kakarot_scripts.constants import (
+    EVM_ADDRESS,
+    NETWORK,
+    RPC_CLIENT,
+    TOKEN_ADDRESSES_DIR,
+    NetworkType,
+)
+from kakarot_scripts.utils.kakarot import deploy as deploy_kakarot
+from kakarot_scripts.utils.kakarot import deploy_and_fund_evm_address
+from kakarot_scripts.utils.kakarot import dump_deployments as dump_evm_deployments
+from kakarot_scripts.utils.kakarot import get_deployments as get_evm_deployments
+from kakarot_scripts.utils.starknet import deploy as deploy_starknet
+from kakarot_scripts.utils.starknet import execute_calls, get_class_hash_at
+from kakarot_scripts.utils.starknet import get_contract as get_contract_starknet
+from kakarot_scripts.utils.starknet import get_deployments as get_starknet_deployments
+from kakarot_scripts.utils.starknet import (
+    get_starknet_account,
+    invoke,
+    register_lazy_account,
+    remove_lazy_account,
+)
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+# %%
+
+
+async def deploy_starknet_token(owner_address: str) -> Any:
+    address = await deploy_starknet("StarknetToken", int(1e18), owner_address)
+    return get_contract_starknet("StarknetToken", address=address)
+
+
+async def deploy_dualvm_token(
+    kakarot_address: str, starknet_token_address: str, deployer_account: Any = None
+) -> Any:
+    dual_vm_token = await deploy_kakarot(
+        "CairoPrecompiles",
+        "DualVmToken",
+        kakarot_address,
+        starknet_token_address,
+        caller_eoa=deployer_account,
+    )
+
+    await invoke(
+        "kakarot",
+        "set_authorized_cairo_precompile_caller",
+        int(dual_vm_token.address, 16),
+        True,
+    )
+    return dual_vm_token
+
+
+async def deploy_dualvm_tokens() -> None:
+    # %% Deploy DualVM Tokens
+    kakarot = get_starknet_deployments()["kakarot"]
+    evm_deployments = get_evm_deployments()
+    tokens = await get_tokens(NETWORK)
+
+    for token in tokens:
+        token_name = token["name"]
+        if token_name not in evm_deployments:
+            await deploy_new_token(token_name, token, kakarot, evm_deployments)
+        else:
+            await verify_and_update_existing_token(
+                token_name, token, kakarot, evm_deployments
+            )
```

Comment:
```suggestion
    for token in tokens:
        token_name = token["name"]
        if token_name not in evm_deployments:
            await deploy_new_token(token_name, token, kakarot, evm_deployments)
        else:
            await verify_and_update_existing_token(
                token_name, token, kakarot, evm_deployments
            )
```

why would you need to extract `token_name` from `token` to eventually give them both to the functions?